packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -170,41 +185,43 @@ export class SCWSigner implements Signer {
       throw standardErrors.provider.unauthorized('Invalid session');
     }
 
-    return decryptContent(content.encrypted, sharedSecret);
+    const response: RPCResponse<T> = await decryptContent(content.encrypted, sharedSecret);
+
+    const metadata = response.data;
+    if (metadata) {
+      this.cacheMetadata(metadata);
```

Comment:
run "updateInternalState" logic, renamed as `cacheMetadata` for all responses received.